lib/hocon/impl/path.rb

```diff
@@ -35,11 +26,67 @@ def self.has_funky_chars?(s)
   end
 
   def initialize(first, remainder)
+    # first: String, remainder: Path
+
     @first = first
     @remainder = remainder
   end
   attr_reader :first, :remainder
 
+  def self.from_string_list(elements)
+    # This method was translated from the Path constructor in the
+    # Java hocon library that has this signature:
+    #   Path(String... elements)
+    #
+    # It figures out what @first and @remainder should be, then
+    # pass those to the ruby constructor
+    if elements.length == 0
+      raise Hocon::ConfigError::ConfigBugOrBrokenError("empty path")
+    end
+
+    new_first = elements.first
+
+    if elements.length > 1
+      pb = Hocon::Impl::PathBuilder.new
+
+      # Skip first element
+      elements.drop(1).each do |element|
+        pb.append_key(element)
+      end
+
+      new_remainder = pb.result
+    else
+      new_remainder = nil
+    end
+
+    self.new(new_first, new_remainder)
+  end
+
+  def self.from_path_list(path_list)
```

Comment:
Can you please add a comment, as in the method above, about the constructor (from the Java version) that this represents?